all-repos.yaml

```diff
@@ -221,3 +221,4 @@
 - https://github.com/sirwart/ripsecrets
 - https://github.com/bagerard/graphviz-dot-hooks
 - https://github.com/omnilib/ufmt
+- https://github.com/phylum-dev/phylum-ci
```

Comment:
this doesn't appear to be configured correctly:

- it does not filter filenames
- it sets `require_serial` though afaict doesn't involve any sort of multiprocessing

Reply:
> * it does not filter filenames

This is on purpose. Phylum supports analyzing dependency lockfiles from a range of languages. Some lockfile types (e.g., Python/pip `requirements.txt`) are ambiguous in that they can be named differently (e.g., `requirements-dev.txt`, `dev-requirements.txt`, etc.) and may or may not contain strict dependencies (e.g., `pyyaml` vs. `pyyaml==5.3.1`). In these cases, it is best to specify an explicit lockfile path.

Instead of trying to capture all possible lockfile names in the `.pre-commit-hooks.yaml` file, the [documentation to configure `.pre-commit-config.yaml`](https://docs.phylum.io/docs/git_precommit#configure-pre-commit-configyaml) specifies the [file control/filter method](https://docs.phylum.io/docs/git_precommit#file-control) to limit the hook being run to only the times when the explicitly identified lockfile has changed, saving execution time.

Further, the list of currently supported lockfile types, which is expanding as new languages are supported, can be viewed [in documentation](https://docs.phylum.io/docs/analyzing-dependencies) or [in source](https://github.com/phylum-dev/phylum-ci/blob/main/src/phylum/constants.py#L31).

> * it sets `require_serial` though afaict doesn't involve any sort of multiprocessing

Upon further review, `require_serial` is not required to be set for the `phylum` hook. However, changing that meant also changing how the script/hook exits. These changes are currently in the [`not_serial` branch](https://github.com/phylum-dev/phylum-ci/tree/not_serial) of the repository and will be merged to `main` after a review...likely this week.